pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -228,53 +228,40 @@ cdef class _Timestamp(ABCTimestamp):
             ots = other
         elif other is NaT:
             return op == Py_NE
-        elif PyDateTime_Check(other):
-            if self.nanosecond == 0:
+        elif PyDateTime_Check(other) or is_datetime64_object(other):
```

Comment:
you don't want to move the nanosecond==0 case up a level? (maybe marginally more clear)
e.g. 
```
elif self.nanosecond == 0 and PyDateTime_Check(...)
    .....
elif PyDatetime_Check or is_datetime64.......
```

Reply:
i considered that, ended up deciding i prefer keeping datetime-scalar-like checks together to mirror what we do elsewhere